cli/src/semgrep/config_resolver.py

```diff
@@ -330,35 +330,34 @@ def parse_config_files(
     return config
 
 
-class ConfigPath:
-    def __init__(self, config_str: str, project_url: Optional[str] = None) -> None:
-        self._config_str = config_str
-        self._project_url = project_url
-
-    def resolve_config(self) -> Dict[str, YamlTree]:
-        """resolves if config arg is a registry entry, a url, or a file, folder, or loads from defaults if None"""
-        start_t = time.time()
-
-        config = parse_config_files(
-            ConfigLoader(self._config_str, self._project_url).load_config()
-        )
-
-        if config:
-            logger.debug(f"loaded {len(config)} configs in {time.time() - start_t}")
-        return config
+def resolve_config(
+    config_str: str, project_url: Optional[str] = None
+) -> Dict[str, YamlTree]:
+    """resolves if config arg is a registry entry, a url, or a file, folder, or loads from defaults if None"""
+    start_t = time.time()
+    config_loader = ConfigLoader(config_str, project_url)
+    config = parse_config_files(config_loader.load_config())
 
-    def __str__(self) -> str:
-        # TODO return the resolved config_path
-        return self._config_str
+    if config:
+        logger.debug(f"loaded {len(config)} configs in {time.time() - start_t}")
+    return config
 
 
 class Config:
-    def __init__(self, valid_configs: Mapping[str, Sequence[Rule]]) -> None:
+    def __init__(
+        self,
+        valid_configs: Mapping[str, Sequence[Rule]],
+        *,
+        with_code_rules: bool = False,
```

Comment:
would it be cleaner to take a list of semgrep_output_v1.product instead of 2 booleans?
so that next time we add a new product we don't need to change this code?

Reply:
I'd prefer being simple and explicit here since we only have 3 products to date (and shouldn't expect to change this more than every ~6 months).

Reply:
But we already started the migration of code using 3 booleans to instead using the new product type. For example in semgrep_output_v1.atd we used to have is_sca_scan, is_code_scan, is_secrets_scan booleans, and now instead we have one field with a list of products.

Reply:
Noted. I can add a TODO here than to address in a future PR. I'll also tag https://github.com/returntocorp/semgrep/blob/develop/cli/src/semgrep/run_scan.py#L159